week-4/Homework/mandatory/2-writers.js

```diff
@@ -40,5 +48,61 @@ let writers = [
 ];
 
 /*
-If you want an extra challenge, only `console.log()` the writers that are alive.
+Exercise 1:
+  Loop through the Array, and for each object, `console.log()` out the sentence and
+  insert corresponding values to the place holder that are indicated in courle braces:
+  "Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
 */
+
+  console.log ('output using array methods filter and mapping');
+  var data = writers.map (writer => {
+  console.log ('Hi,my name is ' + writer.firstName + ' ' + writer.lastName + ' . I am ' + writer.age + ' years old, and work as a '+writer.occupation);
+  });
+
+  console.log ();
+//let me try for in here
+//struggling with for in here as this is an object array its best to use simple for loop 
+  console.log ('output using for loop');
+  for (var indexes=0; indexes< writers.length;indexes++) {
+  console.log ('Hi , my name is ' + writers[indexes].firstName + ' ' + writers[indexes].lastName + ' . I am ' + writers[indexes].age + ' years old, and work as a ' + writers[indexes].occupation);
+}
+
+// for each 
+  console.log ();
+  console.log ('output using for each loop');
+  writers.forEach ((writer) =>
+  console.log(
+    'Hi, my name is ' + writer.firstName +' ' + writer.lastName + ' . I am  '+ writer.age +', and work as a' + writer.occupation
+  )
+)
+/*
+Exercise 2:
+  Only `console.log()` the writers who are in their 40s with us anymore with the sentence:(meaning between 40 and 49)
+  "Writer {firstName} {lastName} died at {age} years old."
+*/
+
+//first use filter array to find writers in their 40's
+  console.log ('Writers who are in their fourties:')
+  var inFourtiesAndDied = writers.filter (lookupAge => lookupAge.age > 40 && lookupAge.age < 50);

```

Comment:
```suggestion
const inFourtiesAndDied = writers.filter (lookupAge => lookupAge.age > 40 && lookupAge.age < 50);
```
Avoid using the `var` keyword